ml/models/logo_detection/detect.py

```diff
@@ -0,0 +1,144 @@
+        imgsz = check_img_size(imgsz, s=stride)  # check img_size
+        return stride, imgsz
+
+
+    def get_dataset(self, source, imgsz, stride):
+        dataset = LoadImages(source, img_size=imgsz, stride=stride)
+        return dataset
+
+
+    def get_names(self, model):
+        names = model.module.names if hasattr(model, 'module') else model.names
+        return names
+
+
+    def run_inference(self, model, device, imgsz, names, half, dataset):
+        inferences = []
+        if device.type != 'cpu':
+                model(torch.zeros(1, 3, imgsz, imgsz).to(device).type_as(next(model.parameters())))  # run once
+
+        t0 = time.time()
+        for path, img, im0s, vid_cap in dataset:
+            img = torch.from_numpy(img).to(device)
+            img = img.half() if half else img.float()  # uint8 to fp16/32
+            img /= 255.0  # 0 - 255 to 0.0 - 1.0
+            if img.ndimension() == 3:
+                img = img.unsqueeze(0)
+
+            # Inference
+            t1 = time_synchronized()
+            with torch.no_grad():   # Calculating gradients would cause a GPU memory leak
+                pred = model(img, augment=self.augment)[0]
+            t2 = time_synchronized()
+
+            # Apply NMS
+            pred = non_max_suppression(pred, self.conf_thres, self.iou_thres, classes=self.classes, agnostic=self.agnostic_nms)
+            t3 = time_synchronized()
+
+            # Process detections
+            for i, det in enumerate(pred):  # detections per image
+                p, s, im0, frame = path, '', im0s, getattr(dataset, 'frame', 0)
+
+                p = Path(p)  # to Path
+                gn = torch.tensor(im0.shape)[[1, 0, 1, 0]]  # normalization gain whwh
+                if len(det):
+                    # Rescale boxes from img_size to im0 size
+                    det[:, :4] = scale_coords(img.shape[2:], det[:, :4], im0.shape).round()
+                    # print(f"BOXES ---->>>> {det[:, :4]}")
+
+                # Print results
+                    for c in det[:, -1].unique():
+                        n = (det[:, -1] == c).sum()  # detections per class
+                        s += f"{n} {names[int(c)]}{'s' * (n > 1)}, "  # add to string
+
+                    # Write results
+                    for *xyxy, conf, cls in reversed(det):
+                        xywh = (xyxy2xywh(torch.tensor(xyxy).view(1, 4)) / gn).view(-1).tolist()  # normalized xywh
+
+                        # Store data into the classes
+                        bbox_coordinates = BoundingBox(x=xywh[0], y=xywh[1], w=xywh[2], h=xywh[3])
+                        inference = LogoDetectionInference(boundingBox=bbox_coordinates, confidence=conf.item())
+                        inferences.append(inference)
+
+                # Print time (inference + NMS)
+                print(f'{s}Done. ({(1E3 * (t2 - t1)):.1f}ms) Inference, ({(1E3 * (t3 - t2)):.1f}ms) NMS')
+
+            print(f'Done. ({time.time() - t0:.3f}s)')
```

Comment:
Small thing, but can we change these `prints` to `logger.info` statements. At the top of the file you can put

```python
import logging
... rest of imports

logging.basicConfig()
logger = logging.getLogger()
logger.setLevel(logging.INFO)
```